detox/android/detox/src/full/java/com/wix/detox/espresso/matcher/RegexMatcher.kt

```diff
@@ -0,0 +1,42 @@
+package com.wix.detox.espresso.matcher
+
+// Returns whether the whole string matches the given `jsRegex`.
+// JS flags has the format of `/<pattern>/<flags>`.
+// Flags can be either:
+// - i: With this flag the search is case-insensitive: no difference between A and a (see the example below).
+// - s: Enables “dotall” mode, that allows a dot . to match newline character \n (covered in the chapter Character classes).
+// - m: Multiline mode (covered in the chapter Multiline mode of anchors ^ $, flag "m").
+// Other flags (e.g. g,u,s) are not supported as they do not have equivalents in Kotlin.
+//
+// - See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
+fun String.isMatch(jsRegex: String): Boolean {
```

Comment:
I think calling it `matchesJSRegex` would be more self-explanatory. `isMatch` is too broad

Reply:
I know I know, that's what I called it in iOS; I take the responsibility. The same comment applies 😅